### PR TITLE
PFM-65 Add teams and roles to profile

### PR DESCRIPTION
**Pull request by SamDudley**

### CODE REVIEW

File: peoplefinder/views/role.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+from django.contrib import messages
```
Comment (SamDudley) [inline peoplefinder/views/role.py:1]: check over for potential refactor

File: peoplefinder/forms/role.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class RoleForm(forms.ModelForm):
+    class Meta:
+        model = TeamMember
+        fields = [
+            "person",
+            "team",
+            "job_title",
+            "head_of_team",
+        ]
+        widgets = {"person": forms.HiddenInput()}
```
Comment (kevinetienne) [inline peoplefinder/forms/role.py:15]: Is there any risk, if a user temper with the person pk in the form?

File: peoplefinder/views/role.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        return render(
+            request, "peoplefinder/components/role-edit.html", context=context
+        )
+
+    def delete(self, request, *args, **kwargs):
+        if self.role:
+            self.role.delete()
+
+        return HttpResponse("")
```
Comment (kevinetienne) [inline peoplefinder/views/role.py:64]: Do you need a separate view to handle the deletion as the view is called `RoleEditView` and should you return a redirect? Do you have a constraint with the js? It looks like this view support both rendering html and responding to an ajax call for deletion.

File: peoplefinder/views/role.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+from django.http import HttpResponse, JsonResponse
+from django.shortcuts import render
+
+from peoplefinder.forms.role import RoleForm
+from peoplefinder.models import Person, TeamMember
+from peoplefinder.services.team import TeamService
+from .base import PeoplefinderView
+
+
+class RoleEditView(PeoplefinderView):
```
Comment (kevinetienne) [inline peoplefinder/views/role.py:11]: Who can edit role, do you need a permission check or see if the current user editing the role is allowed to change the roles?

**Review by kevinetienne — COMMENTED** [review]
Added 3 comments which seems to revolve around the same topic (who can edit/delete roles).

File: peoplefinder/views/role.py (hunk 3 above)
Comment (SamDudley) [reply peoplefinder/views/role.py:64]: I do need to handle deletion and the empty response is for htmx to be able to remove the role from the DOM. I will add a comment explaining the empty response and will rename the view `RoleFormView`.

File: peoplefinder/forms/role.py (hunk 2 above)
Comment (SamDudley) [reply peoplefinder/forms/role.py:15]: Yes there is, I'll add some validation to the post to check this.

File: peoplefinder/views/role.py (hunk 4 above)
Comment (SamDudley) [reply peoplefinder/views/role.py:11]: Added a basic check, the permissions in peoplefinder are interesting and will be implemented in detail at a later date.

**Review by kevinetienne — APPROVED** [review]
I didn't know about `UserPassesTestMixin` :+1:. Looks good to me but you might want to have someone as you mentioned to review the js.

**Pull request merged**